probes/https-probe.js

```diff
@@ -0,0 +1,157 @@
+          that.metricsProbeStart(probeData, httpsReq.method, traceUrl);
+          that.requestProbeStart(probeData, httpsReq.method, traceUrl);
+          aspect.after(res, 'end', probeData, function(obj, methodName, args, probeData, ret) {
+            that.metricsProbeEnd(probeData, httpsReq.method, traceUrl, res, httpsReq);
+            that.requestProbeEnd(probeData, httpsReq.method, traceUrl, res, httpsReq);
+          });
+        }
+      });
+    });
+  }
+  return target;
+};
+
+/*
+ * Custom req.url parser that strips out any trailing query
+ */
+var parse = function(url) {
+  ['?', '#'].forEach(function(separator) {
+    var index = url.indexOf(separator);
+    if (index !== -1) url = url.substring(0, index);
+  });
+  return url;
+};
+
+/*
+ * Ignore requests for URLs which we've been configured via regex to ignore
+ */
+HttpsProbe.prototype.filterUrl = function(req) {
+  var resultUrl = parse(req.url);
+  var filters = this.config.filters;
+  if (filters.length == 0) return resultUrl;
+
+  var identifier = req.method + ' ' + resultUrl;
+  for (var i = 0; i < filters.length; ++i) {
+    var filter = filters[i];
+    if (filter.regex.test(identifier)) {
+      return filter.to;
+    }
+  }
+  return resultUrl;
+};
+
+/*
+ * Lightweight metrics probe for HTTPS requests
+ *
+ * These provide:
+ * 		time:		time event started
+ * 		method:		HTTPS method, eg. GET, POST, etc
+ * 		url:		The url requested
+ * 		duration:	the time for the request to respond
+ */
+
+HttpsProbe.prototype.metricsEnd = function(probeData, method, url, res, httpsReq) {
+  if (probeData && probeData.timer) {
+    probeData.timer.stop();
+    am.emit('https', {
+      time: probeData.timer.startTimeMillis,
+      method: method,
+      url: url,
+      duration: probeData.timer.timeDelta,
+      header: res._header,
+      statusCode: res.statusCode,
+      contentType: res.getHeader('content-type'),
+      requestHeader: httpsReq.headers,
+    });
+  }
+};
+
+/*
+ * Heavyweight request probes for HTTPS requests
+ */
+
+HttpsProbe.prototype.requestStart = function(probeData, method, url) {
+  var reqType = 'https';
+  // Mark as a root request as this happens due to an external event
+  probeData.req = request.startRequest(reqType, url, true, probeData.timer);
+};
+
+HttpsProbe.prototype.requestEnd = function(probeData, method, url, res, httpsReq) {
+  if (probeData && probeData.req)
+    probeData.req.stop({
+      url: url,
+      method: method,
+      requestHeader: httpsReq.headers,
+      statusCode: res.statusCode,
+      header: res._header,
+      contentType: res.getHeader('content-type'),
+    });
+};
+
+/*
+ * Set configuration by merging passed in config with current one
+ */
+HttpsProbe.prototype.setConfig = function(newConfig) {
+  if (typeof newConfig.filters !== 'undefined') {
+    newConfig.filters.forEach(function(filter) {
+      if (typeof filter.regex === 'undefined') {
+        filter.regex = new RegExp(filter.pattern);
+      }
+    });
+  }
```

Comment:
Yes, I think this is ok, it's because there's more than one way of setting config options and this merges them (copied from existing http-probe).  
As a side note it looks like we don't have any tests for URL filtering so that would be good to do.